### PR TITLE
SLCORE-350 Support CSV encoded multi-valued properties

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/MapSettings.java [hunk 1]
```diff
@@ -265,9 +265,7 @@ public Double getDouble(String key) {
         return ArrayUtils.EMPTY_STRING_ARRAY;
       }
 
-      return Arrays.stream(value.split(",", -1)).map(String::trim)
-        .map(s -> s.replace("%2C", ","))
```
Comment (damien-urruty-sonarsource) [inline analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/MapSettings.java:269]: We were doing some kind of URL decode here, and I think we don't do it anymore, was that important ?

File: commons/src/test/java/org/sonarsource/sonarlint/core/commons/log/SonarLintLogTester.java [hunk 2]
```diff
@@ -65,6 +65,15 @@
 
   private final Queue<String> logs = new ConcurrentLinkedQueue<>();
   private final Map<Level, Queue<String>> logsByLevel = new ConcurrentHashMap<>();
+  private final boolean writeToStdOut;
+
+  public SonarLintLogTester(boolean writeToStdOut) {
```
Comment (damien-urruty-sonarsource) [inline commons/src/test/java/org/sonarsource/sonarlint/core/commons/log/SonarLintLogTester.java:70]: I don't see this constructor invoked, you want to keep it to ease debugging later on local machine ?

**Review by damien-urruty-sonarsource — APPROVED** [review]
Some questions, and also you could replace some variable declarations with `var`s.

File: commons/src/test/java/org/sonarsource/sonarlint/core/commons/log/SonarLintLogTester.java (hunk 2 above)
Comment (henryju) [reply commons/src/test/java/org/sonarsource/sonarlint/core/commons/log/SonarLintLogTester.java:70]: Yeah, I used it to investigate a test failure.

File: analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/MapSettings.java (hunk 1 above)
Comment (henryju) [reply analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/sonarapi/MapSettings.java:269]: AFAIK this is a remain of a very very old bug we had in the UI. I don't think we need to keep that anymore.

**Review by damien-urruty-sonarsource — APPROVED** [review]

**Pull request merged**